agent/vendor/github.com/aws/amazon-ecs-agent/ecs-agent/api/resource/ebs_discovery_windows.go

```diff
@@ -16,6 +16,70 @@
 
 package resource
 
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+const (
+	volumeDiscoveryTimeoutDuration = 5 * time.Second
+)
+
 func (api EBSDiscoveryClient) ConfirmEBSVolumeIsAttached(deviceName, volumeID string) error {
+	ctxWithTimeout, cancel := context.WithTimeout(api.ctx, volumeDiscoveryTimeoutDuration)
+	defer cancel()
+	output, err := exec.CommandContext(ctxWithTimeout,
+		"C:\\PROGRAMDATA\\Amazon\\Tools\\ebsnvme-id.exe").CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "failed to run ebsnvme-id.exe: %s", string(output))
+	}
+
+	_, err = parseExecutableOutput(output, volumeID, deviceName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse ebsnvme-id.exe output for volumeID: %s and deviceName: %s",
+			volumeID, deviceName)
+	}
+
+	log.Info(fmt.Sprintf("found volume with volumeID: %s and deviceName: %s", volumeID, deviceName))
+
 	return nil
 }
+
+// parseExecutableOutput parses the output of `ebsnvme-id.exe` and returns the volumeId.
+func parseExecutableOutput(output []byte, candidateVolumeId string, candidateDeviceName string) (string, error) {
+	/* The output of the ebsnvme-id.exe is emitted like the following:
```

Comment:
This seems like a really easily breakable parse. I think I would prefer we do something like:
```
lines := split output by lines
for i := 0; i <len(lines); i++ {
   if lines[i] starts with Disk Number:
      number, id, name, err := parseSet(lines, i)
      if err != nil {
          return "", err
      }
     i = i + 2
      if name == candidateDeviceName {
          return id, nil
      }
  }
  return "", not found
}

func parseSet(lines []string, index int) (string, string, string, error) {
    parse disk number at [i] or error
   parse volume id at [i+1] or error
   parse device name at [i+2] or error
}
```

Reply:
Addressed the comments.